universe/Ship.h

```diff
@@ -1,11 +1,15 @@
 #ifndef _Ship_h_
 #define _Ship_h_
 
+#include "ShipDesign.h"
```

Comment:
That's okay for now.  Moving the `INVALID_DESIGN_ID` out of the `ShipDesign` class to allow making it `extern` would be a good future issue anyway.